src/rules/maxStatementsWithExclusions/maxStatementsWithExclusions.test.js

```diff
@@ -0,0 +1,183 @@
+const { RuleTester } = require('eslint');
+const rule = require('./index');
+const mockCallback = require('./mockCallback');
+
+const ruleTester = new RuleTester({ parserOptions: { ecmaVersion: 2020 } });
+
+const validWithExclusions = [
+  `function Test() {
+    console.log('1');
+    console.log('2');
+    console.log('3');
+    console.log('4');
+    console.log('5');
+  }
+`,
+  ` function Test() {
+    console.log('1');
+    console.log('2');
+    console.log('3');
+    console.log('4');
+    console.log('5');
+    logger.log('6');
+  }
+`,
+  `function Test() {
+    console.log('1');
+    console.log('2');
+    console.log('3');
+    console.log('4');
+    logger.log('5');
+    console.log('6');
+    logger.log('7');
+  }
+`,
+  `function Test() {
+  logger.log('1');
+  logger.log('2');
+  logger.log('3');
+  logger.log('4');
+  logger.log('5');
+  logger.log('6');
+  logger.log('7');
+  logger.log('8');
+}
+`,
+];
+
+const invalidWithExclusions = [
+  `function Test() {
+    console.log('1');
+    console.log('2');
+    console.log('3');
+    console.log('4');
+    console.log('5');
+    console.log('6');
+  }
+`,
+  `function Test() {
+  console.log('1');
+  console.log('2');
+  console.log('3');
+  console.log('4');
+  console.log('5');
+  console.log('6');
+  logger.log('11');
+}
+`,
+];
+
+const optionsWithExclusions = [
+  { max: 5 },
+  { exclusionCallback: require.resolve('./mockCallback') },
+];
+
+ruleTester.run('max-statements-with-exclusions', rule, {
+  valid: validWithExclusions.map((code) => ({
+    code,
+    options: optionsWithExclusions,
+  })),
+
+  invalid: invalidWithExclusions.map((code) => {
+    const max = optionsWithExclusions[0].max;
+    return {
+      code,
+      options: optionsWithExclusions,
+      errors: [
+        {
+          message: `Function 'Test' has too many statements that haven't been excluded (${
+            max + 1
+          }). Maximum allowed is ${max}.`,
+        },
+      ],
+    };
+  }),
+});
+
+const validWithoutExclusions = [
+  `function Test() {
+    console.log('1');
+    console.log('2');
+    console.log('3');
+    console.log('4');
+    console.log('5');
+  }
+`,
+  ` function Test() {
+    console.log('1');
+    console.log('2');
+    console.log('3');
+    console.log('4');
+    console.log('5');
+  }
+`,
+  `function Test() {
+    console.log('1');
+    console.log('2');
+    console.log('3');
+    console.log('4');
+    logger.log('5');
+  }
+`,
+  `function Test() {
+  logger.log('1');
+  logger.log('2');
+  logger.log('3');
+  logger.log('4');
+  logger.log('5');
+}
+`,
+];
+
+const invalidWithoutExclusions = [
+  `function Test() {
+    console.log('1');
+    console.log('2');
+    console.log('3');
+    console.log('4');
+    console.log('5');
+    console.log('6');
+  }
+`,
+  `function Test() {
+  console.log('1');
+  console.log('2');
+  console.log('3');
+  console.log('4');
+  console.log('5');
+  logger.log('6');
+}
+`,
+];
```

Comment:
Extra test cases for both valid and invalid scenarios:
- function declaration
- function expression
- instance class methods
- static class methods
- constructor
- arrow function
- arrow function inside class (as a field)
- nested functions (combinations of the above)
- exported functions (variations of the above)

Reply:
the above tests added - and some test helpers to produce the various function types - seems to work well - you make think it's overkill but I didn't want to have to keep writing out all the scenarios.  I think I made it reusable for other tests - at least for statements anyway

Linting run and all tests passing.

Please can you check and authorise to run the checks workflow